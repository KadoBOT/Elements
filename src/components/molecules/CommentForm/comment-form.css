.comment-form {
  border-bottom: 4px solid #ec0f50;
  padding-bottom: 100px;
}

.comment-form__row {
  display: flex;
}

.comment-form__row--text-area {
  display: flex;
  margin-top: 70px;
}

.comment-form__row--submit {
  text-align: center;
  padding: 30px;
}

.comment-form__row > label:first-child {
  margin-right: 70px;
}

.comment-form__row > label:not(:first-child) {
  margin-left: 70px;
}
