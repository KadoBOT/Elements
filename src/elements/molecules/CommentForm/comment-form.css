.comment-form {
  border-bottom: 0.25vw solid #ec0f50;
  padding-bottom: 5vw;
}

.comment-form__row {
  display: flex;
  padding: 2vw 0px;
}

.comment-form__row--text-area {
  display: flex;
  margin-top: 3.5vw;
}

.comment-form__row--submit {
  text-align: center;
  padding: 1.5vw;
}

.comment-form__row > label:first-child {
  margin-right: 3.5vw;
}

.comment-form__row > label:not(:first-child) {
  margin-left: 3.5vw;
}
