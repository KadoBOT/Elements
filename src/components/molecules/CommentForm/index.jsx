import React, {PropTypes} from 'react'

import { Button, Input, TextArea } from '../../atoms'
import './comment-form.css'

const CommentForm = ({comment, errors, onSubmit, onChange}) => {
  const submitForm = () => {
    onSubmit()
  }

  return(
    <div className="comment-form">
      <div className="comment-form__row">
        <Input
          errors={errors.name}
          isRequired
          label="Name"
          name="name"
          placeholder="Place your name here..."
          onChange={onChange}
          type="text"
          value={comment.name}
        />
        <Input
          errors={errors.email}
          isRequired
          label="Email"
          name="email"
          placeholder="username@company.com"
          onChange={onChange}
          type="email"
          value={comment.email}
        />
      </div>
      <div className="comment-form__row--text-area">
        <TextArea
          errors={errors.comment}
          label="Comment"
          isRequired
          name="comment"
          placeholder="Place your comment here..."
          onChange={onChange}
          value={comment.comment}
        />
      </div>
      <div className="comment-form__row--submit" >
        <Button onSubmit={submitForm}>
          Submit Comment
        </Button>
      </div>
    </div>
  )
}

CommentForm.propTypes = {
  comment: PropTypes.any.isRequired,
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default CommentForm
