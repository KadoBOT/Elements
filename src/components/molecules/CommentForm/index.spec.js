import React from 'react'
import { shallow, render, mount } from 'enzyme';

import CommentForm from './'

it('should render the CommentForm', () => {
    const wrapper = shallow(
        <CommentForm
          comment={{name: '', email: '', comment: ''}}
          errors={{name: [], email: [], comment: []}}
          onSubmit={jest.fn()}
          onChange={jest.fn()}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
