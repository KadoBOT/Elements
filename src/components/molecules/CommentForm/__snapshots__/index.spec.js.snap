exports[`test should render the CommentForm 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="comment-form">
    <div
        className="comment-form__row">
        <Input
            errors={Array []}
            isRequired={true}
            label="Name"
            name="name"
            onChange={[Function]}
            placeholder="Place your name here..."
            type="text"
            value="" />
        <Input
            errors={Array []}
            isRequired={true}
            label="Email"
            name="email"
            onChange={[Function]}
            placeholder="username@company.com"
            type="email"
            value="" />
    </div>
    <div
        className="comment-form__row--text-area">
        <TextArea
            errors={Array []}
            isRequired={true}
            label="Comment"
            name="comment"
            onChange={[Function]}
            placeholder="Place your comment here..."
            value="" />
    </div>
    <div
        className="comment-form__row--submit">
        <Button
            onSubmit={[Function]}>
            Submit Comment
        </Button>
    </div>
</div>,
  "nodes": Array [
    <div
      className="comment-form">
      <div
            className="comment-form__row">
            <Input
                  errors={Array []}
                  isRequired={true}
                  label="Name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Place your name here..."
                  type="text"
                  value="" />
            <Input
                  errors={Array []}
                  isRequired={true}
                  label="Email"
                  name="email"
                  onChange={[Function]}
                  placeholder="username@company.com"
                  type="email"
                  value="" />
      </div>
      <div
            className="comment-form__row--text-area">
            <TextArea
                  errors={Array []}
                  isRequired={true}
                  label="Comment"
                  name="comment"
                  onChange={[Function]}
                  placeholder="Place your comment here..."
                  value="" />
      </div>
      <div
            className="comment-form__row--submit">
            <Button
                  onSubmit={[Function]}>
                  Submit Comment
            </Button>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CommentForm
        comment={
                Object {
                        "comment": "",
                        "email": "",
                        "name": "",
                      }
        }
        errors={
                Object {
                        "comment": Array [],
                        "email": Array [],
                        "name": Array [],
                      }
        }
        onChange={[Function]}
        onSubmit={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "comment": Object {
            "comment": "",
            "email": "",
            "name": "",
          },
          "errors": Object {
            "comment": Array [],
            "email": Array [],
            "name": Array [],
          },
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="comment-form">
          <div
                    className="comment-form__row">
                    <Input
                              errors={Array []}
                              isRequired={true}
                              label="Name"
                              name="name"
                              onChange={[Function]}
                              placeholder="Place your name here..."
                              type="text"
                              value="" />
                    <Input
                              errors={Array []}
                              isRequired={true}
                              label="Email"
                              name="email"
                              onChange={[Function]}
                              placeholder="username@company.com"
                              type="email"
                              value="" />
          </div>
          <div
                    className="comment-form__row--text-area">
                    <TextArea
                              errors={Array []}
                              isRequired={true}
                              label="Comment"
                              name="comment"
                              onChange={[Function]}
                              placeholder="Place your comment here..."
                              value="" />
          </div>
          <div
                    className="comment-form__row--submit">
                    <Button
                              onSubmit={[Function]}>
                              Submit Comment
                    </Button>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="comment-form">
          <div
                    className="comment-form__row">
                    <Input
                              errors={Array []}
                              isRequired={true}
                              label="Name"
                              name="name"
                              onChange={[Function]}
                              placeholder="Place your name here..."
                              type="text"
                              value="" />
                    <Input
                              errors={Array []}
                              isRequired={true}
                              label="Email"
                              name="email"
                              onChange={[Function]}
                              placeholder="username@company.com"
                              type="email"
                              value="" />
          </div>
          <div
                    className="comment-form__row--text-area">
                    <TextArea
                              errors={Array []}
                              isRequired={true}
                              label="Comment"
                              name="comment"
                              onChange={[Function]}
                              placeholder="Place your comment here..."
                              value="" />
          </div>
          <div
                    className="comment-form__row--submit">
                    <Button
                              onSubmit={[Function]}>
                              Submit Comment
                    </Button>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CommentForm
    comment={
        Object {
            "comment": "",
            "email": "",
            "name": "",
          }
    }
    errors={
        Object {
            "comment": Array [],
            "email": Array [],
            "name": Array [],
          }
    }
    onChange={[Function]}
    onSubmit={[Function]} />,
}
`;
